packages/pds/src/app-view/api/app/bsky/unspecced.ts

```diff
@@ -0,0 +1,66 @@
+import { Server } from '../../../../lexicon'
+import { FeedKeyset, composeFeed } from './util/feed'
+import { paginate } from '../../../../db/pagination'
+import AppContext from '../../../../context'
+import { FeedRow, FeedItemType } from '../../../services/feed'
+import { sql } from 'kysely'
+
+// THIS IS A TEMPORARY UNSPECCED ROUTE
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.unspecced.getPopular({
+    auth: ctx.accessVerifier,
+    handler: async ({ params, auth }) => {
+      const { limit, before } = params
+      const requester = auth.credentials.did
+      const db = ctx.db.db
+      const { ref } = db.dynamic
+
+      console.log('here')
+
+      const feedService = ctx.services.appView.feed(ctx.db)
+
+      let postsQb = ctx.db.db
+        .with('vote_counts', (qb) =>
+          qb
+            .selectFrom('vote')
+            .where('vote.direction', '=', 'up')
+            .groupBy('vote.subject')
+            .select([sql`count(*)`.as('count'), 'vote.subject']),
+        )
+        .selectFrom('post')
+        .innerJoin('vote_counts', 'vote_counts.subject', 'post.uri')
+        .leftJoin('repost', (join) =>
+          join
+            .on('repost.creator', '=', 'did:plc:ea2eqamjmtuo6f4rvhl3g6ne')
+            .onRef('repost.subject', '=', 'post.uri'),
+        )
```

Comment:
Just a note that this works well for one repost creator but if we allow multiple it could create dupes in the result set.

Reply:
yup yup  
i'll add a note in the code 👍